index.js

```diff
@@ -24,14 +25,26 @@ module.exports = function frameguard (options) {
   }
 
   if (directive === 'ALLOW-FROM') {
-    if (!isString(domain)) {
-      throw new Error('X-Frame: ALLOW-FROM requires a second parameter')
+    if (!isString(domain) && !isArray(domain)) {
+      throw new Error('X-Frame: ALLOW-FROM requires a domain parameter')
     }
-    directive = 'ALLOW-FROM ' + domain
+  }
+
+  function header (req) {
+    var headerValue = directive
+    if (directive === 'ALLOW-FROM' && isString(domain)) {
```

Comment:
sure, why not! 😄 
I usually dislike infinite chaining blocks... from my perspective this is simpler.

When you start chaining if/else blocks you usually end with things like this:

```js

if (a) {
  if (b && !c) {
    // a && b && !c
  } else if (b && c) {
    // a && b && c
  }
  // (a && b) || (a && !b)
} else {
  // !a
}
```

while you can simplify the analysis by:

```js
if (a && b && !c) {
  // a && b && !c
}

if (a && b && c) {
  // a && b && c
}

if (a) {
  // a && b || a && !b
}

// !a
```

I'll update the PR later